src/cpu/aarch64/acl_winograd_convolution.hpp

```diff
@@ -0,0 +1,177 @@
+        acl_wino_obj_->bia_tensor.allocator()->init(acp.bia_info);
+
+        // clang-format off
+        acl_wino_obj_->conv.configure(
+            &acl_wino_obj_->src_tensor,
+            &acl_wino_obj_->wei_tensor,
+            acp.with_bias ? &acl_wino_obj_->bia_tensor : nullptr,
+            &acl_wino_obj_->dst_tensor,
+            acp.padstride_info,
+            acp.act_info,
+            true); // to support 5x5, 7x7 filter shapes in addition to 3x3
+        // clang-format on
+
+        return status::success;
+    }
+
+    acl_obj_t<arm_compute::NEWinogradConvolutionLayer> &get_acl_obj() const {
+        return *acl_wino_obj_;
+    }
+
+    DNNL_DISALLOW_COPY_AND_ASSIGN(acl_wino_resource_t);
+
+private:
+    std::unique_ptr<acl_obj_t<arm_compute::NEWinogradConvolutionLayer>>
+            acl_wino_obj_;
+}; // acl_wino_resource_t
+
+struct acl_wino_convolution_fwd_t : public primitive_t {
+    struct pd_t : public cpu_convolution_fwd_pd_t {
+        pd_t(const convolution_desc_t *adesc, const primitive_attr_t *attr,
+                const typename pd_t::base_class *hint_fwd_pd)
+            : cpu_convolution_fwd_pd_t(adesc, attr, hint_fwd_pd), acp_() {}
+
+        DECLARE_COMMON_PD_T(
+                "wino:acl", acl_wino_convolution_fwd_t, USE_GLOBAL_SCRATCHPAD);
+
+        status_t init(engine_t *engine) {
+            bool ok = true && is_fwd()
+                    && utils::one_of(desc()->alg_kind,
+                            alg_kind::convolution_auto,
+                            alg_kind::convolution_winograd)
+                    && expect_data_types(data_type::f32, data_type::f32,
+                            data_type::f32, data_type::f32, data_type::f32)
+                    && attr()->has_default_values(
+                            primitive_attr_t::skip_mask_t::post_ops,
+                            data_type::f32)
+                    && !has_zero_dim_memory() && post_ops_ok();
+            if (!ok) return status::unimplemented;
+
+            auto conf_status = acl_convolution_utils::init_conf_wino(acp_,
+                    src_md_, weights_md_, dst_md_, bias_md_, *desc(), *attr());
+            if (conf_status != status::success) return status::unimplemented;
+
+            set_default_alg_kind(alg_kind::convolution_winograd);
+
+            // Number of threads in Compute Library is set by OMP_NUM_THREADS
+            // dnnl_get_max_threads() == OMP_NUM_THREADS
+            arm_compute::Scheduler::get().set_num_threads(
+                    dnnl_get_max_threads());
+
+            // TODO: remove dependence on scratchpad memory
+            // Using user provided memory for the biases currently segfaults
+            if (acp_.with_bias) {
+                auto scratchpad = scratchpad_registry().registrar();
+                const size_t bia_mem_sz_
+                        = acp_.bia_info.tensor_shape()[0] * sizeof(data_t);
+                scratchpad.template book<data_t>(
```

Comment:
For the templated `book` routine of scratchpad I don't think we have to multiply `bia_mem_sz_` by `sizeof(data_t)`

Reply:
Seems like we missed that in `acl_gemm_convolution.hpp` as well.